src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1887,11 +1887,33 @@ void CodeGen::genAvxOrAvx2Intrinsic(GenTreeHWIntrinsic* node)
 
     if ((op1 != nullptr) && !op1->OperIsList())
     {
+        op1Reg = op1->gtRegNum;
         genConsumeOperands(node);
     }
 
     switch (intrinsicId)
     {
+        case NI_AVX2_ConvertToDouble:
+        {
+            assert(op2 == nullptr);
+            if (op1Reg != targetReg)
+            {
+                instruction ins = HWIntrinsicInfo::lookupIns(intrinsicId, baseType);
+                emit->emitIns_R_R(ins, emitTypeSize(targetType), targetReg, op1Reg);
+            }
+            break;
+        }
+
+        case NI_AVX2_ConvertToInt32:
+        case NI_AVX2_ConvertToUInt32:
+        {
+            assert(op2 == nullptr);
+            assert((baseType == TYP_INT) || (baseType == TYP_UINT));
```

Comment:
This assert seems a little bit improper because `ConvertToInt32` only accepts `int` and `ConvertToUInt32` only accepts `uint`, but not very important.